ecosystem/gstreamer_plugin/gst_mtl_st40p_tx.c

```diff
@@ -350,17 +392,19 @@ static gboolean gst_mtl_st40p_tx_sink_event(GstPad* pad, GstObject* parent,
   ret = GST_EVENT_TYPE(event);
 
   switch (GST_EVENT_TYPE(event)) {
-    case GST_EVENT_SEGMENT:
-      if (!sink->tx_handle) {
-        GST_ERROR("Tx handle not initialized");
-        return FALSE;
+    case GST_EVENT_STREAM_START:
+      if (sink->async_session_create) {
+        thread_data = malloc(sizeof(GstMtlSt40pTxThreadData));
+        thread_data->sink = sink;
+        pthread_create(&sink->session_thread, NULL,
+                       gst_mtl_st40p_tx_session_create_thread, thread_data);
+      } else {
+        ret = gst_mtl_st40p_tx_session_create(sink);
```

Comment:
do you know how much later GST_EVENT_STREAM_START is send after gst_mtl_st40p_tx_start() was called? Are we ok with introducing this delay to the case when async_session_create == false?

Reply:
we absolutly are not 
I was under the impression that this signal is being "sent" much faster, it is noooot after the packets starts flowing its being sent 
i though that leaving this in signals would be fine but i don't think so no more will move him